wordpress-mobile/WordPress-iOS
# [This Week Widget] add data handling

Ref #13044 

This adds data handling for the 'This Week' Widget.
- When a Period view is accessed in the WP app, 'This Week' data is stored in `ThisWeekData.plist`.
- The widget loads and saves data from/to the plist.
- The widget loads site information from `UserDefaults`, and uses it to fetch stat data.

The data is temporarily logged to the console to facilitate testing.

To test:
- Run the WP app.
- Go to stats > Widgets > Use this site.
- Add the 'This Week' widget to your Today view.
- In order to see the console logging, at this point you'll need to run the `WordPressThisWeekWidget` target in Xcode.

<img width="400" alt="target" src="https://user-images.githubusercontent.com/1816888/70488167-0dbb4180-1ab5-11ea-97d0-eddbb6bb049b.png">

- When you run the target, the `siteUrl` and `statsValues` are logged in the console.

```
🔴 siteUrl:  <site_url>
🔴 statsValues:  Optional(WordPressThisWeekWidget.ThisWeekWidgetStats(days
[WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-10 07:00:00 +0000, viewsCount: 7, dailyChange: -427), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-09 07:00:00 +0000, viewsCount: 434, dailyChange: 361), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-08 07:00:00 +0000, viewsCount: 73, dailyChange: 40), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-07 07:00:00 +0000, viewsCount: 33, dailyChange: -224), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-06 07:00:00 +0000, viewsCount: 257, dailyChange: -141), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-05 07:00:00 +0000, viewsCount: 398, dailyChange: 39), 
WordPressThisWeekWidget.ThisWeekWidgetDay(date: 2019-12-04 07:00:00 +0000, viewsCount: 359, dailyChange: 153)]))
```

- Verify the stats match the Days' chart data for each day.

<img width="400" alt="overview_chart_days" src="https://user-images.githubusercontent.com/1816888/70488193-26c3f280-1ab5-11ea-82a8-2ed9c2ac51aa.png">

- Return the app and switch sites used for the widgets.
- Return to the widget.
- Verify the logged stats match the newly selected site.

PR submission checklist:
